### PR TITLE
Feature/warn only crystal version

### PR DESCRIPTION
**Pull request by beta-ziliani**
This PR changes the semantics of the `crystal:` field to be a "maintainers claim it works in these versions of the compiler", warning in case the current version is not listed. An empty field means `*`, as coded in PR #493 by @oprypin.

It is an implementation (by @bcardiff) of the proposal I made as a comment in PR #493.

### CODE REVIEW

File: src/cli.cr
Comment (oprypin) [inline src/cli.cr:47]: This flag must not become invalid, else we're just adding another breakage on top.

https://github.com/search?q=%22ignore+crystal+version%22&type=code

File: src/cli.cr
Comment (Sija) [inline src/cli.cr:47]:
```suggestion
      # TODO: remove in the future
      opts.on("--ignore-crystal-version", "Kept for compatibility, to be removed in the future.") {  }
```

File: docs/shard.yml.adoc
Comment (straight-shoota) [inline docs/shard.yml.adoc:85]:
```suggestion
have been determined shards checks all of them are expected to work with
the current crystal version. If not, a warning appears for the offending
dependencies. The resolved versions are installed and can be used at your 
```

File: docs/shard.yml.adoc
Comment (straight-shoota) [inline docs/shard.yml.adoc:97]:
```suggestion
_x.y.z_ is interpreted as _">= x.y.z"_
```

File: src/cli.cr
Comment (straight-shoota) [inline src/cli.cr:48]:
```suggestion
      opts.on("--ignore-crystal-version", "Has no effect. Kept for compatibility, to be removed in the future.") {  }
```

**Review by straight-shoota — COMMENTED** [review]
I have a few suggestions to improve wording. Apart from that, it lookgs good to me.

**Comment by straight-shoota** [comment]
It seems this needs a formatter run.

**Comment by beta-ziliani** [comment]
I think we're good to go, once the CI passes let's merge this (pinging @straight-shoota)

**Review by straight-shoota — APPROVED** [review]
I'm not sure the reverted `crystal: x.y.z` semantics are the final word, but we can figure that out afterwards. Since it's just a warning now, the implications are not much of an issue.

**Comment by luislavena** [comment]
Hello @beta-ziliani @bcardiff, possible a new release of shards with this change is made?

Not asking packages of Crystal 1.0.1 to be released, but at least having shards updates so impacted folks can use or point users to that manual upgrade until a new distribution of Crystal is released.

Thank you.

**Comment by oprypin** [comment]
(literally speaking, it should be possible, [as mentioned here](https://github.com/crystal-lang/shards/pull/493#issuecomment-818899381))

**Pull request merged**
